lib/osc.dart

```diff
@@ -5,37 +5,50 @@ import 'dart:io';
 import 'package:udp/udp.dart';
 import 'message.dart';
 
+// abstract interface class Sender {
+//   Future<int> send(List<int> data, Endpoint remoteEndpoint);
+//   Stream<Datagram?> asStream({Duration? timeout});
+//   close();
+// }
+
 class Conn {
-  late final String _remoteHost;
-  late final int _remotePort;
-  late final int _localPort;
   late final Endpoint _dest;
-  late UDP _sender;
+  late final UDP _sender;
+
+  Conn({required Endpoint dest, required UDP sender})
+      : _dest = dest,
+        _sender = sender;
 
-  Conn({required remoteHost, remotePort = 10023, localPort = 10023})
-      : _remoteHost = remoteHost,
-        _remotePort = remotePort,
-        _localPort = localPort {
-    _init();
+  factory Conn.withUDP({required Endpoint dest, required UDP sender}) {
+    return Conn(dest: dest, sender: sender);
   }
 
-  Future _init() async {
-    _dest = Endpoint.multicast(InternetAddress(_remoteHost),
-        port: Port(_remotePort));
-    _sender = await UDP.bind(Endpoint.any(port: Port(_localPort)));
+  static Future<Conn> initUDP(
+      {required remoteHost, remotePort = 10023, localPort = 10023}) async {
+    final dest =
+        Endpoint.unicast(InternetAddress(remoteHost), port: Port(remotePort));
+    final sender = await UDP.bind(Endpoint.any(port: Port(localPort)));
+
+    return Conn.withUDP(dest: dest, sender: sender);
   }
```

Comment:
I still can't understand the purpose of the factory constructor here. Why not just call the default constructor from the static function?
```
Conn({required Endpoint dest, required UDP sender})
    : _dest = dest,
      _sender = sender;

static Future<Conn> initUDP({required String remoteHost, int remotePort = 10023, int localPort = 10023}) async {
    final dest = Endpoint.unicast(InternetAddress(remoteHost), port: Port(remotePort));
    final sender = await UDP.bind(Enpoint.any(port: Port(localPort)));
    return Conn(dest: dest, sender: sender);
}
```